packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -312,40 +339,48 @@ export class ProjectDataSource {
     this._specWatcher.on('all', onProjectFileSystemChange)
   }
 
-  async defaultSpecFileName () {
-    const getDefaultFileName = (testingType: TestingType) => `cypress/${testingType}/${testingType === 'e2e' ? 'spec' : 'ComponentName'}.cy.${this.ctx.lifecycleManager.fileExtensionToUse}`
+  async defaultSpecFileName (): Promise<string> {
+    const defaultFilename = `${this.ctx.coreData.currentTestingType === 'e2e' ? 'spec' : 'ComponentName'}.cy.${this.ctx.lifecycleManager.fileExtensionToUse}`
+    const defaultPathname = path.join('cypress', this.ctx.coreData.currentTestingType ?? 'e2e', defaultFilename)
 
     try {
       if (!this.ctx.currentProject || !this.ctx.coreData.currentTestingType) {
-        return getDefaultFileName('e2e')
+        throw new Error('Failed to get default spec filename, missing currentProject/currentTestingType')
```

Comment:
Do you want to move this outside of the try/catch? We could cut out the middle man and just return defaultPathname otherwise.

Reply:
oh woof I didn't even notice that this is the same thing